### PR TITLE
KEYCLOAK-847 Add documentation for the step-up mechanism

### PR DESCRIPTION
**Pull request by CorneliaLahnsteiner**
Documentation for the step-up mechanism. Related PR: https://github.com/keycloak/keycloak/pull/7897

### CODE REVIEW

**Review by mposolda — DISMISSED** [review]
@CorneliaLahnsteiner Thanks! I am asking also @andymunro (our documentation guru) to review this

@andymunro Do you please have a chance to review this?

File: server_admin/topics/authentication/flows.adoc [hunk 1]
```diff
@@ -160,7 +160,7 @@ To illustrate the creation of flows, this section describes creating an advanced
 .The common part with the browser flow
 image:images/Passwordless-browser-login-common.png[]
 +
-. Click *Actions for the *Forms* execution.
+. Click *Actions* for the *Forms* execution.
 . Select *Add execution*.
 . Select *Username Form* from the drop-down list.
```
Comment (andymunro) [inline server_admin/topics/authentication/flows.adoc:165]: We now call it an "item list" per IBM.

File: server_admin/topics/authentication/flows.adoc [hunk 2]
```diff
@@ -224,6 +224,129 @@ Creating an advanced flow such as this can have side effects. For example, if yo
...
+This section describes how to create advanced browser login flow using the step-up mechanism. The purpose of step-up authentication is to allow access to clients or resources based on a specific authentication level of a user.
+
+.Procedure
+. Click *Authentication* in the menu.
+. Click the *Flows* tab.
+. Click *New*.
+. Enter `Browser Incl Step up Mechanism` as an alias.
+. Click *Save*.
+. Click *Add execution*.
+. Select *Cookie* from the drop-down list.
```
Comment (andymunro) [inline server_admin/topics/authentication/flows.adoc]: We now call it an "item list" per IBM.

File: server_admin/topics/authentication/flows.adoc [hunk 3]
```diff
@@ -224,6 +224,129 @@ Creating an advanced flow such as this can have side effects. For example, if yo
...
+Now you configure the flow for the first authentication level.
+
+. Click *Actions* for the *Auth Flow*.
+. Click *Add flow*.
+. Enter `1st Condition Flow` as an alias.
+. Click *Save*.
+. Click *Conditional* for the *1st Condition Flow* authentication type to set its requirement to conditional.
+. Click *Actions* for the *1st Condition Flow*.
+. Click *Add execution*.
+. Select *Conditional - Level Of Authentication* from the drop-down list.
```
Comment (andymunro) [inline server_admin/topics/authentication/flows.adoc]: We now call it an "item list" per IBM.

File: server_admin/topics/authentication/flows.adoc [hunk 4]
```diff
@@ -224,6 +224,129 @@ Creating an advanced flow such as this can have side effects. For example, if yo
...
+. Enter `1` for the Level of Authentication (LoA).
+. Set Store LoA in user session *ON*.
+. Click *Save*
++
+.Configure the condition for the first authentication level
+image:images/authentication-step-up-condition-1.png[]
+
+. Click *Actions* for the *1st Condition Flow*.
+. Click *Add execution*.
+. Select *Username Password Form* from the drop-down list.
```
Comment (andymunro) [inline server_admin/topics/authentication/flows.adoc]: We now call it an "item list" per IBM.

File: server_admin/topics/authentication/flows.adoc [hunk 5]
```diff
@@ -224,6 +224,129 @@ Creating an advanced flow such as this can have side effects. For example, if yo
...
+Now you configure the flow for the second authentication level.
+
+. Click *Actions* for the *Auth Flow*.
+. Click *Add flow*.
+. Enter `2nd Condition Flow` as an alias.
+. Click *Save*.
+. Click *Conditional* for the *2nd Condition Flow* authentication type to set its requirement to conditional.
+. Click *Actions* for the *2nd Condition Flow*.
+. Click *Add execution*.
+. Select *Conditional - Level Of Authentication* from the drop-down list.
```
Comment (andymunro) [inline server_admin/topics/authentication/flows.adoc]: We now call it an "item list" per IBM.

File: server_admin/topics/authentication/flows.adoc [hunk 6]
```diff
@@ -224,6 +224,129 @@ Creating an advanced flow such as this can have side effects. For example, if yo
...
+. Enter `2` for the Level of Authentication (LoA).
+. Set Store LoA in user session *OFF*.
+. Click *Save*
++
+.Configure the condition for the second authentication level
+image:images/authentication-step-up-condition-2.png[]
+
+. Click *Actions* for the *2nd Condition Flow*.
+. Click *Add execution*.
+. Select *OTP Form* from the drop-down list.
```
Comment (andymunro) [inline server_admin/topics/authentication/flows.adoc]: We now call it an "item list" per IBM.

File: server_admin/topics/authentication/flows.adoc [hunk 7]
```diff
@@ -224,6 +224,129 @@ Creating an advanced flow such as this can have side effects. For example, if yo
...
+. Click *Save*.
+. Click *Required* for the *OTP Form* authentication type to set its requirement to required.
+
+Finally, change the bindings.
+
+. Click the *Bindings* tab.
+. Click the *Browser Flow* drop-down list.
```
Comment (andymunro) [inline server_admin/topics/authentication/flows.adoc]: We now call it an "item list" per IBM.

File: server_admin/topics/authentication/flows.adoc [hunk 8]
```diff
@@ -224,6 +224,129 @@ Creating an advanced flow such as this can have side effects. For example, if yo
...
+. Select *Browser Incl Step up Mechanism* from the drop-down list.
```
Comment (andymunro) [inline server_admin/topics/authentication/flows.adoc]: We now call it an "item list" per IBM.

File: server_admin/topics/clients/oidc/con-advanced-settings.adoc [hunk 9]
```diff
@@ -88,3 +88,13 @@ The procedure to select the algorithm is:
 . Open the client's *Settings* tab.
```
Comment (andymunro) [inline server_admin/topics/clients/oidc/con-advanced-settings.adoc:88]: Should be "Click" not "Open"

File: server_admin/topics/clients/oidc/con-advanced-settings.adoc [hunk 10]
```diff
@@ -88,3 +88,13 @@ The procedure to select the algorithm is:
...
 . Open *Fine Grain OpenID Connect Configuration*.
 . Select the algorithm from *ID Token Encryption Content Encryption Algorithm* pulldown menu.
```
Comment (andymunro) [inline server_admin/topics/clients/oidc/con-advanced-settings.adoc:90]: Is this the same as a drop-down menu? If so it should be an "item list" per IBM.

File: server_admin/topics/clients/oidc/con-advanced-settings.adoc [hunk 11]
```diff
@@ -88,3 +88,13 @@ The procedure to select the algorithm is:
...
+
+[[_mapping-acr-to-loa]]
+*ACR to Level of Authentication (LoA) Mapping*
+
+In the advanced settings of a client you can define which `Authentication Context Class Reference (ACR)` value is mapped to which `Level of Authentication (LoA)`. The ACR can be any value, whereas the LoA must be numeric. +
```
Comment (andymunro) [inline server_admin/topics/clients/oidc/con-advanced-settings.adoc]: Add a comma after client.
Looks an extra plus sign after numeric.

**Review by andymunro — CHANGES_REQUESTED** [review]
Looks good @mposolda.   If you want to hold off on making changes of drop-down list to item list, we can do that separately in another PR for the entire Admin Guide.

File: server_admin/topics/authentication/flows.adoc (hunk 3 above)
Comment (CorneliaLahnsteiner) [reply server_admin/topics/authentication/flows.adoc]: Done.

File: server_admin/topics/authentication/flows.adoc (hunk 4 above)
Comment (CorneliaLahnsteiner) [reply server_admin/topics/authentication/flows.adoc]: Done.

File: server_admin/topics/authentication/flows.adoc (hunk 5 above)
Comment (CorneliaLahnsteiner) [reply server_admin/topics/authentication/flows.adoc]: Done.

File: server_admin/topics/authentication/flows.adoc (hunk 6 above)
Comment (CorneliaLahnsteiner) [reply server_admin/topics/authentication/flows.adoc]: Done.

File: server_admin/topics/authentication/flows.adoc (hunk 7 above)
Comment (CorneliaLahnsteiner) [reply server_admin/topics/authentication/flows.adoc]: Done.

File: server_admin/topics/authentication/flows.adoc (hunk 8 above)
Comment (CorneliaLahnsteiner) [reply server_admin/topics/authentication/flows.adoc]: Done.

File: server_admin/topics/clients/oidc/con-advanced-settings.adoc (hunk 10 above)
Comment (CorneliaLahnsteiner) [reply server_admin/topics/clients/oidc/con-advanced-settings.adoc:90]: As this is text that was not added or changed by me, I would prefer not to change it as I am not sure if the pulldown menu is a dropdown menu.

File: server_admin/topics/clients/oidc/con-advanced-settings.adoc (hunk 11 above)
Comment (CorneliaLahnsteiner) [reply server_admin/topics/clients/oidc/con-advanced-settings.adoc]: Done

File: server_admin/topics/clients/oidc/con-advanced-settings.adoc (hunk 9 above)
Comment (CorneliaLahnsteiner) [reply server_admin/topics/clients/oidc/con-advanced-settings.adoc:88]: As this is text that was not added or changed by me, I would prefer not to change it as I am not sure if Click or Open is the correct wording.

File: server_admin/topics/authentication/flows.adoc [hunk 12]
```diff
@@ -224,6 +224,136 @@ Creating an advanced flow such as this can have side effects. For example, if yo
...
+            "id_token": {
+                "acr": {
+                    "essential": true,
+                    "values": ["gold"]
+                }
+            }
+        }
+----
+
+NOTE: To request the acr_vale as text (such as `gold`) instead of a numeric value, you have to configure the mapping between the ACR and the LoA at the client level. For configuration see <<_mapping-acr-to-loa,ACR to LoA Mapping>>)
```
Comment (fidgi) [inline server_admin/topics/authentication/flows.adoc:334]: typo : ... the acr_val**u**e as text ...

**Pull request merged**